### PR TITLE
feat(wallet): additional activity search criteria

### LINKED ISSUE #2174: Activity search should allow filtering on node alias
## Detailed Description

Activity search should allow filtering on node alias, especially since we display node alias in the list view.

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Enhance activity search to support searching on destination addresses, node pubkey, alias, and preimage.

## Motivation and Context:

Fix #2174

## How Has This Been Tested?

Manually

## Types of changes:

Enhancement

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [ ] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: renderer/reducers/activity.js [hunk 1]
```diff
@@ -141,12 +140,28 @@ const transactionsSelector = state => transactionsSelectors.transactionsSelector
...
 
+/**
+ * Augment payments with additional data.
+ */
+const decoratedPaymentsSelector = createSelector(
+  paymentsSelector,
+  nodesSelector,
+  (payments, nodes) =>
+    payments.map(payment => {
+      const destPubKey = payment.path[payment.path.length - 1]
```
Comment (korhaliv) [inline renderer/reducers/activity.js]: Is `payment.path` always set?

File: renderer/reducers/activity.js [hunk 2]
```diff
@@ -279,7 +294,11 @@ const applySearch = (data, searchText) => {
         item =>
           (item.tx_hash && item.tx_hash.includes(searchText)) ||
           (item.payment_hash && item.payment_hash.includes(searchText)) ||
-          (item.payment_request && item.payment_request.includes(searchText))
+          (item.payment_preimage && item.payment_preimage.includes(searchText)) ||
```
Comment (korhaliv) [inline renderer/reducers/activity.js]: consider extracting this into a helper to reduce potential typo errors. Smth like
`const matches = (prop) => item[props] &&  item[prop].includes(searchText)`
and then 
`['text_has','payment_hash','payment_preimage',...].some(matches)`

File: renderer/reducers/activity.js (hunk 1 above)
Comment (mrfelton) [reply renderer/reducers/activity.js]: It should be, but I'ave added some safeguards around this just incase.

File: renderer/reducers/activity.js (hunk 2 above)
Comment (mrfelton) [reply renderer/reducers/activity.js]: Yes, good idea. Have refactored this.

**Pull request merged**